ordered_product/tests.py

```diff
@@ -0,0 +1,147 @@
+
+
+@pytest.fixture
+def supplier_product0(product0, supplier0):
+    return SupplierProduct(suppler_product_id=1,
+                           qr_code=product0,
+                           user_name=supplier0,
+                           price=10,
+                           quantity=50)
+
+
+@pytest.fixture
+def delivery_location0(supplier0):
+    return DeliveryLocation(user_name=supplier0,
+                            location="Kiryat Shmona",
+                            date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def ordered_product0(delivery_location0, client0, supplier_product0):
+    return OrderedProduct(
+        delivery_location_id=delivery_location0,
+        user_name=client0,
+        supplier_product_id=supplier_product0,
+        quantity=3)
+
+
+class TestOrderedProductModel():
+    @pytest.mark.django_db()
+    def test_save_and_delete_ordered_product(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.all()
+        ordered_product0.delete_ordered_product()
+        assert ordered_product0 not in OrderedProduct.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_id(1000))
+        filt = OrderedProduct.filter_id(ordered_product0.id)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_by_user_name(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_user_name(""))
+        filt = OrderedProduct.filter_user_name(ordered_product0.user_name)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_by_supplier_product_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_supplier_product_id(-1))
+        filt = OrderedProduct.filter_supplier_product_id(ordered_product0.supplier_product_id)
+        assert ordered_product0 in list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_delivery_location_id(self, delivery_location0, supplier_product0, ordered_product0):
+        delivery_location0.save()
+        supplier_product0.save()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_delivery_location_id(-1))
```

Comment:
this should be in its own test function i guess

Reply:
Why? I'm don't understand